code/game/mecha/equipment/tools/tools.dm

```diff
@@ -594,89 +599,77 @@
 	energy_drain = 100
 	range = 0
 	var/health_boost = 2
-	var/datum/global_iterator/pr_repair_droid
 	var/icon/droid_overlay
 	var/list/repairable_damage = list(MECHA_INT_TEMP_CONTROL,MECHA_INT_TANK_BREACH)
 
-	New()
-		..()
-		pr_repair_droid = new /datum/global_iterator/mecha_repair_droid(list(src),0)
-		pr_repair_droid.set_delay(equip_cooldown)
-		return
-
-	Destroy()
-		qdel(pr_repair_droid)
-		pr_repair_droid = null
-		return ..()
-
-	attach(obj/mecha/M as obj)
-		..()
-		droid_overlay = new(src.icon, icon_state = "repair_droid")
-		M.overlays += droid_overlay
-		return
-
-	destroy()
-		chassis.overlays -= droid_overlay
-		..()
-		return
-
-	detach()
-		chassis.overlays -= droid_overlay
-		pr_repair_droid.stop()
-		..()
-		return
+/obj/item/mecha_parts/mecha_equipment/repair_droid/Destroy()
+	STOP_PROCESSING(SSprocessing, src)
+	return ..()
 
-	get_equip_info()
-		if(!chassis) return
-		return "<span style=\"color:[equip_ready?"#0f0":"#f00"];\">*</span>&nbsp;[src.name] - <a href='?src=\ref[src];toggle_repairs=1'>[pr_repair_droid.active()?"Dea":"A"]ctivate</a>"
+/obj/item/mecha_parts/mecha_equipment/repair_droid/attach(obj/mecha/M as obj)
+	..()
+	droid_overlay = new(src.icon, icon_state = "repair_droid")
+	M.overlays += droid_overlay

```

Comment:
This should probably be `add_overlay()`, but make sure that if there's any other overlays that they use it too.